crosschain-resolver/contracts/ITargetResolver.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+interface ITargetResolver{
+    function setTarget(
+        bytes32 node, address target
+    ) external;
+
+    function getTarget(
```

Comment:
We should expose a public function without `offset`; `offset` is only needed internally for recursive calls.